### PR TITLE
PDP comments

### PR DESCRIPTION
**Pull request by ecl1pseo**
- Added comments mutations for PDP
- Added token to headers
- Fixed adding product to wishlist and cart
- Replaced **apollo-boost** with **apollo-client**

### CODE REVIEW

**Review by INR4GE — APPROVED** [review]

**Review by DmytroDidukh — APPROVED** [review]

File: src/pages/product-details/comments/comments.js [hunk 1]
```diff
@@ -1,74 +1,258 @@
...
+    setComment,
+    clearFields,
+    filterText,
+    validateField
+  } = useValidation();
+
   const [rate, setRate] = useState(0);
 
+  const { _id, email: userEmail, firstName: userName, purchasedProduct } =
+    userData || {};
```
Comment (Alan18081) [inline src/pages/product-details/comments/comments.js:63]: We should not try any destructuring if no `userData` is found so `userData || {}` is pretty useless, isn't it?

File: src/redux/products/products.reducer.js [hunk 2]
```diff
@@ -193,6 +199,32 @@ const productsReducer = (state = initialState, action = {}) => {
       ...state,
       productLoading: action.payload
     };
+  case SET_RATE:
```
Comment (Alan18081) [inline src/redux/products/products.reducer.js:202]: I think it is better to move all the redux data related to the comments to the separate reducer `Comments`.

File: src/redux/snackbar/snackbar.actions.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {
+  SET_SNACKBAR_MESSAGE,
+  SET_SNACKBAR_SEVERITY,
+  SET_SNACKBAR_STATUS
+} from './snackbar.types';
+
+const setSnackBarStatus = (newSnackBarStatus) => ({
+  type: SET_SNACKBAR_STATUS,
```
Comment (Alan18081) [inline src/redux/snackbar/snackbar.actions.js:8]: Do we really need to have three separate actions for setting the snackbar? As I understand we don't update the message inside it, but just display a new one and close old. If it is true it is better to have only two actions: `showSnackbar` and `closeSnackbar`

File: src/pages/product-details/comments/comments.js (hunk 1 above)
Comment (ecl1pseo) [reply src/pages/product-details/comments/comments.js:63]: `userData ` for convenience has value of `null `in redux, so i need this condition

File: src/redux/snackbar/snackbar.actions.js (hunk 3 above)
Comment (ecl1pseo) [reply src/redux/snackbar/snackbar.actions.js:8]: We update also type(error or success) and message of snackbar, not only status, so i guess it's better to have separate actions for all this stuff

File: src/pages/product-details/comments/comments-item/comments-item.js [hunk 4]
```diff
@@ -1,17 +1,37 @@
...
+      updatingComment: Products.updatingComment,
+      language: Language.language,
+      userEmail: User.email
+    })
+  );
+
+  const { name, images } = user;
+
+  const [isEditable, setEditable] = useState(false);
+  const [modal, setModal] = useState(false);
```
[the remark this answers is not in the document]
Comment (ecl1pseo) [reply src/pages/product-details/comments/comments-item/comments-item.js:34]: renamed

File: src/redux/products/products.reducer.js (hunk 2 above)
Comment (ecl1pseo) [reply src/redux/products/products.reducer.js:202]: moved

File: src/redux/comments/comments.reducer.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	SET_RATE,
+	SET_COMMENTS_LOADING,
+	SET_UPDATING_COMMENT
+} from './comments.types';
+
+export const initialState = {
+	commentsLoading: false,
+	updatingComment: null,
+	comments: [],
+	rate: 0
```
Comment (Alan18081) [inline src/redux/comments/comments.reducer.js:12]: Why do we need this rate if I have seen the same one in the local state of the component?

File: src/pages/product-details/comments/comments.js (hunk 1 above)
Comment (Alan18081) [reply src/pages/product-details/comments/comments.js]: I don't think that you should perform this operation at all if `userData` is not provided

File: src/redux/comments/comments.reducer.js (hunk 5 above)
Comment (ecl1pseo) [reply src/redux/comments/comments.reducer.js:12]: This rate is for summary product rate value from the backend(all votes), and rate in the local state is for input, when user is leaving the comment

**Pull request merged**